ext/opcache/Optimizer/dfa_pass.c

```diff
@@ -925,6 +928,7 @@ static int zend_dfa_optimize_jmps(zend_op_array *op_array, zend_ssa *ssa)
 					}
 					break;
 				case ZEND_SWITCH_STRING:
+				case ZEND_MATCH_STRING:
 					if (opline->op1_type == IS_CONST) {
 						zval *zv = CT_CONSTANT_EX(op_array, opline->op1.constant);
 						if (Z_TYPE_P(zv) != IS_STRING) {
```

Comment:
I don't see any obvious bugs.

For this branch in particular, I think that if something is a ZEND_SWITCH_STRING, it would have to go to the first case (case by case IS_EQUAL comparisons), but if something was a ZEND_MATCH_STRING, it could safely go to the last case (the default case) instead?

(Both are correct, but the alternative is more efficient and I think it should eliminate more dead code)

```
							removed_ops++;
							MAKE_NOP(opline);
							opline->extended_value = 0;
							take_successor_ex(ssa, block_num, block, block->successors[0]);
```

Reply:
I suppose this is irrelevant now that the `IS_IDENTICAL`/`JMPNZ` chain was removed, no?

Reply:
Unfortunately the following doesn't get optimized:

```php
function test1(string $value) {
    echo match($value) {
        1, 2, 3, 4, 5 => 'a',
        default => 'b',
    };
}

test1(1);
```

```
0000 CV0($value) = RECV 1
0001 MATCH_LONG CV0($value) 1: 0002, 2: 0002, 3: 0002, 4: 0002, 5: 0002, default: 0004
0002 T1 = QM_ASSIGN string("a")
0003 JMP 0005
0004 T1 = QM_ASSIGN string("b")
0005 ECHO T1
0006 RETURN null
```

But that has more to do with the fact that `$value` is a `CV` and not a `CONST`.